### ISSUE TITLE
Quantile regression: first quartile greater than median

### ISSUE THREAD

**Issue by ludovicgauvin** [catboost/catboost#1362]
Hi,

I tried quantile regression using CatBoostRegressor. I would like to have information such as "according to our estimate there is 50% chance that y will be between Q1 and Q3" where Q1 is the first quartile (25%) and Q3 the third quartile (75%). I have to estimate 3 models (median, Q1, and Q3). However, sometimes (~ 2% of the time) predicted Q1 is greater than Q2 or even greater than Q3 and/or Q2 is greater than Q3.

I tried to set `learning_rate=0.01` which is ~twice greater than the automatically calculated one in order to increase the probability that there is a point where over-fitting is detected (`od_type="Iter"`, `od_wait=100` and `iterations = 150000`). Finally, `leaf_estimation_method` is set to `'Exact'`.

Overall, I still get sometimes the issue (~ 0.7% of test sample). However, it does not make sense... Is there a way to be 100% sure that the median is always between Q1 and Q2?

Many thanks,
Ludo

**Comment by Evgueni-Petrov-aka-espetrov** [comment]
In general, there is no recipe to make three different models agree all the time.
Model analysis tools, e.g. shap values, may sometimes explain counter-intuitive predictions and/or find anomalies in the data.

**Comment by ludovicgauvin** [comment]
Thank you for your response.

Of course the same value of predictor may affect differently different quantiles, which the main point of interest of quantile regressions I think.

It is clearly a accuracy issue I think.
I could try to implement some kind of "decay learning_rate" using strategy suggested by @annaveronika in this issue: https://github.com/catboost/catboost/issues/979

**Comment by datenzauberai** [comment]
Maybe a workaround: you could try predicting the median first, include it for the Q1 and Q3 predictions and set a monotonic constraint on it.

**Comment by ludovicgauvin** [comment]
Thank you for your response.
I understand how to impose a monotonic constraint on the impact of features but I do not get how to impose the type of constraint you are talking about.
